client/asset/interface.go

```diff
@@ -1017,6 +1027,47 @@ type TicketBuyer interface {
 	TicketPage(scanStart int32, n, skipN int) ([]*Ticket, error)
 }
 
+// TransactionType is type type of transaction made by a wallet.
+type TransactionType uint16
+
+const (
+	Send TransactionType = iota
+	Receive
+	Swap
+	Redeem
+	Refund
+	CreateBond
+	RedeemBond
+	ApproveToken
+	Acceleration
+)
+
+// WalletTransaction represents a transaction that was made by a wallet.
+type WalletTransaction struct {
+	Type           TransactionType `json:"type"`
+	ID             dex.Bytes       `json:"id"`
+	AmtIn          uint64          `json:"amtIn"`
+	AmtOut         uint64          `json:"amtOut"`
+	Fees           uint64          `json:"fees"`
+	BlockNumber    uint64          `json:"blockNumber"`
+	UpdatedBalance uint64          `json:"updatedBalance"`
+	// AdditionalData contains asset specific information, i.e. nonce
+	// for ETH.
+	AdditionalData map[string]string `json:"additionalData"`
+}
+
+// WalletHistorian is a wallet that is able to retrieve the history of all
+// transactions it has made.
+type WalletHistorian interface {
+	// TxHistory returns all the transactions a wallet has made. If refID
+	// is nil, then transactions starting from the most recent are returned
+	// (past is ignored). If past is true, the transactions prior to the
+	// refID are returned, otherwise the transactions after the refID are
+	// returned. n is the number of transactions to return. If n is <= 0,
+	// all the transactions will be returned.
+	TxHistory(n int, refID *dex.Bytes, past bool) ([]*WalletTransaction, error)
+}
```

Comment:
If there are a lot of transactions, this could be pretty big. Maybe add a hard limit?

Reply:
I think the front ends should handle that. I'll update the RPC server to have a default of 10. Would something go wrong if too many are requested?